Add configure check for rpmatch() and 
creates a compatable macro if it is not provided by the C library.

This is needed for uclibc since it does not have the above function
implemented.

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Index: rpm-5.4.0/configure.ac
===================================================================
--- rpm-5.4.0.orig/configure.ac	2012-05-31 15:40:05.985990822 -0700
+++ rpm-5.4.0/configure.ac	2012-05-31 15:41:15.601995311 -0700
@@ -800,7 +800,7 @@
     ftok getaddrinfo getattrlist getcwd getdelim getline getmode getnameinfo dnl
     getpassphrase getxattr getwd iconv inet_aton lchflags lchmod lchown dnl
     lgetxattr lsetxattr lutimes madvise mempcpy mkdtemp mkstemp mtrace dnl
-    posix_fadvise posix_fallocate putenv realpath regcomp __secure_getenv dnl
+    posix_fadvise posix_fallocate putenv realpath regcomp rpmatch __secure_getenv dnl
     setattrlist setenv setlocale setmode setxattr dnl
     sigaddset sigdelset sigemptyset sighold sigrelse sigpause dnl
     sigprocmask sigsuspend sigaction dnl
Index: rpm-5.4.0/system.h
===================================================================
--- rpm-5.4.0.orig/system.h	2012-05-31 15:40:05.997991504 -0700
+++ rpm-5.4.0/system.h	2012-05-31 15:42:04.941997731 -0700
@@ -410,6 +410,14 @@
 #endif /* defined(__GNUC__) */
 #endif	/* HAVE_MCHECK_H */
 
+#ifndef HAVE_RPMATCH
+#define rpmatch(line) \
+	( (line == NULL)? -1 : \
+	  (*line == 'y' || *line == 'Y')? 1 : \
+	  (*line == 'n' || *line == 'N')? 0 : \
+	  -1 )
+#endif
+
 /* Retrofit glibc __progname */
 #if defined __GLIBC__ && __GLIBC__ >= 2
 #if __GLIBC_MINOR__ >= 1
